Tests/MongoSwiftTests/ServerSelectionTests.swift

```diff
@@ -104,4 +104,24 @@ final class ServerSelectionTests: MongoSwiftTestCase {
             expect(serverDescription.averageRoundTripTimeMS).to(equal(test.newAverageRoundTripTimeMS))
         }
     }
+
+    func testReadPreferenceValidation() throws {
+        var readPreference = ReadPreference.primary
+        readPreference.tagSets = [["tag": "set"]]
+        let topology = TopologyDescription(type: .single, servers: [])
+        expect(try topology.findSuitableServers(readPreference: readPreference, heartbeatFrequencyMS: 0))
+            .to(throwError(errorType: MongoError.InvalidArgumentError.self))
+
+        readPreference.tagSets = [[:]]
+        expect(try topology.findSuitableServers(readPreference: readPreference, heartbeatFrequencyMS: 0))
+            .toNot(throwError())
+
+        readPreference.tagSets = nil
+        expect(try topology.findSuitableServers(readPreference: readPreference, heartbeatFrequencyMS: 0))
+            .toNot(throwError())
+    }
+
+    // TODO: Implement the following tests when server selection is integrated into the driver:
+    // read preferences are correctly passed to mongos
+    // operationCount-based selection selection within latency window mongos prose test
```

Comment:
these are the [two](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection-tests.rst#passing-readpreference-to-mongos) [tests](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection-tests.rst#prose-test) referenced here, both require driver integration so the server selection implementation can't test them yet

Reply:
could you file a ticket we can reference here?